### PR TITLE
Publish multi-arch container images

### PR DESCRIPTION
**Pull request by krishvoor**
This PR introduces a PR, to build & publish
ARM64 & AMD64 container images to quay.io/cloud-bulldozer/netperf

## Type of change

- [ ] Refactor
- [X] New feature
- [ ] Bug fix
- [ ] Optimization
- [ ] Documentation Update

## Description

This PR introduces changes to build, publish container images (ARM64 &  AMD64) to quay.io/cloud-bulldozer/netperf repository

## Related Tickets & Documents

- Related Issue #
- Closes #

## Checklist before requesting a review

- [ ] I have performed a self-review of my code.
- [ ] If it is a core feature, I have added thorough tests.

## Testing
- Please describe the System Under Test.
- Please provide detailed steps to perform tests related to this code change.
- How were the fix/results from this change verified? Please provide relevant screenshots or results.

### CODE REVIEW

**Comment by krishvoor** [comment]
@jtaleric Some initial thoughts I jotted together from my prior experiences on publishing a multi-arch image.
Though this is WIP, can you take a look/review?

**Comment by jtaleric** [comment]
Should we have this be a net-new action versus adding to the goreleaser action?

**Comment by jtaleric** [comment]
> I wanted this to be under the same umbrella/file to have a logical release flow, if you think we shd bifurcate these actions and maintain another file, sure can do it.

Well these are two different paths.

GoReleaser- Builds the binary for k8s-netperf
containers/Dockerfile - is the netperf binary, which k8s-netperf sets as the container image for the client/server.

**Comment by jtaleric** [comment]
is this still a WIP?

File: .github/workflows/build-container.yml [hunk 1]
```diff
@@ -0,0 +1,67 @@
+name: Release k8s-netperf container image
+
+on:
+  push:
+    tags:
+      - "*" # triggers only if new tag version is pushed
```
Comment (jtaleric) [inline .github/workflows/build-container.yml:6]: Do we want to trigger when a change is made to the container file versus when a tag is made?

Comment (krishvoor) [reply .github/workflows/build-container.yml:6]: > Do we want to trigger when a change is made to the container file versus when a tag is made?

For every new release/tag  - we should also "publish" a container image along with a binary as an artifact.

If we trigger for every change, we will end up with container images that might only be consumed unless it is for
dev purposes, so having it as a gating criteria for every change might not be relevant - I am open for thoughts on the same.

Comment (jtaleric) [reply .github/workflows/build-container.yml:6]: I think you are missing my point, if the Dockerfile doesn't change don't rebuild. If the Dockerfile for the tag didn't change, don't rebuild. We shouldn't be rebuilding all too often.

Comment (krishvoor) [reply .github/workflows/build-container.yml:6]: > I think you are missing my point, if the Dockerfile doesn't change don't rebuild. If the Dockerfile for the tag didn't change, don't rebuild. We shouldn't be rebuilding all too often.

I just revisited the code base, I was under the impression we are shipping "k8s-netperf" binaries as well.
It now makes sense, we will keep the changes only when we are

Comment (krishvoor) [reply .github/workflows/build-container.yml]: Updated - ptal.

**Review by jtaleric — APPROVED** [review]
lgtm

**Pull request merged**